Dockerfile

```diff
@@ -0,0 +1,27 @@
+# Use variable base image, such that we can also build for other base images, like alpine.
+ARG BASE_IMAGE=debian:buster-slim
+
+FROM golang:1.16-buster as build
+
+# Set build environment
+ENV GOOS=linux
+ENV GOARCH=amd64
+ENV CGO_ENABLED=0
```

Comment:
Aren't these lines superfluous? They all contain the default values (for this platform).

Reply:
Yeah I thought so too, but if I use `alpine` as base image (which a lot of people do), then it complained about `CGO`. And I always use debian buster to build, so it is not related to the build process itself.

The `GOOS` and `GOARCH` can be removed I think indeed.

Reply:
CGO is enabled by [default](https://pkg.go.dev/cmd/cgo). The `alpine` image is so small, that it does not have native bindings for CGO, so that causes the issues. Therefore, I think it is better to build without CGO, such that we get a binary that works everywhere (just like we do in the regular pipeline).